### PR TITLE
[G4-Process] Contribute to upstream repo

### PR DESCRIPTION
**Pull request by ziyue-pan**
**What I have done**:

- add `main.go` as the entry
- fix G4-Process' model (passed compilation and automigration of GORM)

### CODE REVIEW

File: pkg/process/model.go [hunk 1]
```diff
@@ -14,40 +15,44 @@ type Department struct {
...
 type Registration struct {
-	ID         uint `gorm:"primaryKey"`
-	Doctor     account.Doctor
-	Patient    account.Patient
-	Department Department
-	Date       time.Time
-	HalfDay    HalfDayEnum            // TODO: a validator for registration, only half day is allowed
-	Status     RegistrationStatusEnum `gorm:"default:'committed'"`
+	ID           uint `gorm:"primaryKey"`
+	DoctorID     uint
```
Comment (AsterNighT) [inline pkg/process/model.go]: Hmm so you choose not to use foreign key? That's ok.

File: pkg/router/router.go [hunk 2]
```diff
@@ -1,12 +1,12 @@
 package router
 
 import (
+	"github.com/AsterNighT/software-engineering-backend/pkg/cases"
```
Comment (AsterNighT) [inline pkg/router/router.go:6]: You may need to format the files.

**Review by AsterNighT — CHANGES_REQUESTED** [review]

**Comment by AsterNighT** [comment]
Rest LGTM

File: pkg/process/model.go (hunk 1 above)
Comment (ziyue-pan) [reply pkg/process/model.go]: It should be the foreign key.

I'll fix it.

Comment (ziyue-pan) [reply pkg/process/model.go]: fixed

File: pkg/router/router.go (hunk 2 above)
Comment (ziyue-pan) [reply pkg/router/router.go:6]: done

**Comment by ziyue-pan** [comment]
@AsterNighT I fixed the reference key problems. Now the database is like this (i didn't migrate `chats`)

image

It seems that the `cases` and `doctors` also treat id as normal uint rather than the reference key. I wonder if it is a problem.

**Comment by ziyue-pan** [comment]
Besides, I labeled some attributes as `swaggerignore:true` to refine the API.

**Comment by AsterNighT** [comment]
Hmmm your files are still not formated. I would suggect you try `golangci-lint` locally.

**Comment by ziyue-pan** [comment]
> Hmmm your files are still not formated. I would suggect you try `golangci-lint` locally.

ok i reformat my files in the latest commit

**Comment by AsterNighT** [comment]
You see the imports should be sorted with goimports, but they are not now. I wonder. Could you run goimports directly?

**Comment by AsterNighT** [comment]
pkg/router/router.go should be like

```
"net/http"
"github.com/AsterNighT/software-engineering-backend/pkg/cases"
"github.com/AsterNighT/software-engineering-backend/pkg/chat"
```

not 

```
"github.com/AsterNighT/software-engineering-backend/pkg/cases"
"github.com/AsterNighT/software-engineering-backend/pkg/chat"
"net/http"
```

**Comment by ziyue-pan** [comment]
OK. I found the problem with my Goland preferences. I was using `gofmt` rather than `goimports`. Now I solved the problem

**Comment by AsterNighT** [comment]
emm that does not seem to work, it appears that your IDE is using spaces rather than tabs for indent.

**Comment by ziyue-pan** [comment]
> emm that does not seem to work, it appears that your IDE is using spaces rather than tabs for indent.

OK fixed in the latest commit

**Comment by AsterNighT** [comment]
Great! LGTM

**Review by AsterNighT — APPROVED** [review]

**Pull request merged**